components/security/authentication.rst

```diff
@@ -267,5 +267,55 @@ in) is correct, you can use::
         $user->getSalt()
     );
 
+Authentication Events
+---------------------
+
+The security component provides 4 related authentication events:
+
+===============================  ================================================  =========================================================================
+Name                             Event Constant                                    Argument Passed to the Listener
+===============================  ================================================  =========================================================================
+security.authentication.success  ``AuthenticationEvents::AUTHENTICATION_SUCCESS``  :class:`Symfony\Component\Security\Core\Event\AuthenticationEvent`
+security.authentication.failure  ``AuthenticationEvents::AUTHENTICATION_FAILURE``  :class:`Symfony\Component\Security\Core\Event\AuthenticationFailureEvent`
+security.interactive_login       ``SecurityEvents::INTERACTIVE_LOGIN``             :class:`Symfony\Component\Security\Http\Event\InteractiveLoginEvent`
+security.switch_user             ``SecurityEvents::SWITCH_USER``                   :class:`Symfony\Component\Security\Http\Event\SwitchUserEvent`
+===============================  ================================================  =========================================================================
+
+Authentication Success and Failure Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a provider authenticates the user, a ``security.authentication.success``
+event is dispatched.  Likewise, when no providers authenticate the user,
+a ``security.authentication.failure`` event is dispatched.  You
+could listen on the ``security.authentication.failure`` event, for example,
+in order to log failed login attempts.
+
+.. tip::
+
+    One of the authentication events is always triggered when a request points
+    to a secured area.
```

Comment:
Is this true? I see these events thrown only in `AuthenticationProviderManager`. Is this called if no authentication listener is activated on a request - like if I'm already logged in, but am not at `/login_check` (assuming form login is my only method)? Also, if `anonymous` is activated, does that cause `security.authentication.success` to be dispatched on every request?

This stuff is tricky - all the more reason why I'd like to merge this PR.